packages/material-ui/src/Input/Input.js

```diff
@@ -74,12 +74,41 @@ export const styles = theme => {
     formControl: {
       'label + &': {
         marginTop: 16,
+
+        '&$outlined': {
+          marginTop: 0,
+        },
       },
     },
     /* Styles applied to the root element if the component is focused. */
     focused: {},
     /* Styles applied to the root element if `disabled={true}`. */
     disabled: {},
+    /* Styles applied to the root element if `startAdornment` is provided. */
+    adornedStart: {},
+    /* Styles applied to the root element if `endAdornment` is provided. */
+    adornedEnd: {},
+    /* Styles applied to the root element if `variant="outlined"`. */
+    outlined: {
+      width: '280px',
+      alignItems: 'center',
+
+      '&$multiline': {
+        height: 'auto',
+        overflow: 'hidden',
+
+        // These values are needed to prevent us from
+        // overrunning the notched outline (including label)
+        paddingBottom: '2px',
+        paddingTop: '6px',
+      },
+      '&$adornedStart': {
+        paddingLeft: '14px',
+      },
+      '&$adornedEnd': {
+        paddingRight: '8px',
```

Comment:
Shouldn't it be symmetric? 14px
The IconButton taking more space than needed problem can be solved on the `IconButton` side.